### PR TITLE
[feat][feat] delta neutral reinvest

### PR DESCRIPTION
**Pull request by sora-alpaca**

### CODE REVIEW

File: contracts/8.10/protocol/DeltaNeutralVaultConfig.sol [hunk 1]
```diff
@@ -16,6 +16,7 @@ pragma solidity 0.8.10;
 import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
 
 import "./interfaces/IDeltaNeutralVaultConfig.sol";
+import "./interfaces/ISwapRouter.sol";
```
Comment (alpanaca) [inline contracts/8.10/protocol/DeltaNeutralVaultConfig.sol:19]: Not used

File: contracts/8.10/protocol/DeltaNeutralVaultConfig.sol [hunk 2]
```diff
@@ -26,27 +27,35 @@ contract DeltaNeutralVaultConfig is IDeltaNeutralVaultConfig, OwnableUpgradeable
...
   /// @dev Errors
   error InvalidSetSwapRoute();
   error LeverageLevelTooLow();
   error TooMuchFee(uint256 _depositFeeBps, uint256 _mangementFeeBps);
 
+  error InvalidSwapRouter();
+  error InvalidReinvestPath();
+  error InvalidReinvestPathLength();
+
   struct SwapRoute {
```
Comment (alpanaca) [inline contracts/8.10/protocol/DeltaNeutralVaultConfig.sol:59]: Need this?

File: test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts [hunk 3]
```diff
@@ -3500,4 +3155,252 @@ describe("DeltaNeutralVault", () => {
...
+        const stableVaultFairLaunchPoolId = await stableVault.fairLaunchPoolId();
+        const assetVaultFairLaunchPoolId = await assetVault.fairLaunchPoolId();
+        console.log("stableVaultFairLaunchPoolId", stableVaultFairLaunchPoolId);
+        console.log("assetVaultFairLaunchPoolId", assetVaultFairLaunchPoolId);
+
+        let currentBlock = await TimeHelpers.latestBlockNumber();
+        console.log("currentBlock before claim", currentBlock);
+
+        //TODO not sure need to accPerShare everytime?
+        await fairLaunch.massUpdatePools();
```
Comment (alpanaca) [inline test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts:3329]: I don't think we need this.

File: contracts/8.10/protocol/DeltaNeutralVault.sol [hunk 4]
```diff
@@ -80,16 +82,12 @@ contract DeltaNeutralVault is ERC20Upgradeable, ReentrancyGuardUpgradeable, Owna
   }
 
   /// @dev constants
+  uint256 private constant MAX_BPS = 10000;
```
Comment (jr-alpaca) [inline contracts/8.10/protocol/DeltaNeutralVault.sol:85]: uint64

File: contracts/8.10/protocol/DeltaNeutralVault.sol [hunk 5]
```diff
@@ -80,16 +82,12 @@ contract DeltaNeutralVault is ERC20Upgradeable, ReentrancyGuardUpgradeable, Owna
...
+  uint256 private constant PRICE_AGE_SECOND = 1800; // 30 mins
```
Comment (jr-alpaca) [inline contracts/8.10/protocol/DeltaNeutralVault.sol:86]: remove this line

File: contracts/8.10/protocol/DeltaNeutralVault.sol [hunk 6]
```diff
@@ -397,6 +406,56 @@ contract DeltaNeutralVault is ERC20Upgradeable, ReentrancyGuardUpgradeable, Owna
...
+    uint256 _alpacaBefore = IERC20Upgradeable(alpacaToken).balanceOf(address(this));
+
+    _claim(IVault(stableVault).fairLaunchPoolId());
+    _claim(IVault(assetVault).fairLaunchPoolId());
+
+    uint256 _alpacaAfter = IERC20Upgradeable(alpacaToken).balanceOf(address(this));
+
+    // 2. collect alpaca bounty
+    uint256 _bounty = ((_alpacaBountyBps) * (_alpacaAfter - _alpacaBefore)) / MAX_BPS;
+    SafeERC20Upgradeable.safeTransfer(IERC20Upgradeable(alpacaToken), config.getTreasuryAddr(), _bounty);
```
Comment (jr-alpaca) [inline contracts/8.10/protocol/DeltaNeutralVault.sol:434]: IERC20Upgradeable(alpacaToken).safeTransfer(config.getTreasuryAddr(), _bounty);

File: contracts/8.10/protocol/DeltaNeutralVault.sol [hunk 7]
```diff
@@ -660,10 +724,9 @@ contract DeltaNeutralVault is ERC20Upgradeable, ReentrancyGuardUpgradeable, Owna
     IERC20Upgradeable(assetToken).safeApprove(_vault, 0);
   }
 
-  /// @notice Claim Alpaca reward of stable vault and asset vault
-  function claim() external returns (uint256, uint256) {
-    uint256 rewardStableVault = _claim(IVault(stableVault).fairLaunchPoolId());
-    uint256 rewardAssetVault = _claim(IVault(assetVault).fairLaunchPoolId());
+  /// @notice withdraw alpaca to receiver address
+  function withdrawAlpaca(address _to, uint256 _amount) external onlyOwner {
```
Comment (jr-alpaca) [inline contracts/8.10/protocol/DeltaNeutralVault.sol:727]: remove this function

**Review by jr-alpaca — APPROVED** [review]

**Pull request merged**